RealmSwift/Tests/TestUtils.mm

```diff
@@ -50,6 +50,28 @@ void RLMAssertThrows(XCTestCase *self, dispatch_block_t block, NSString *name, N
     }
 }
 
+void RLMAssertThrowsWithReasonMatching(XCTestCase *self, dispatch_block_t block, NSString *regexString, NSString *message, NSString *fileName, NSUInteger lineNumber) {
```

Comment:
Note that this is copied from Realm Objective-C's Swift helpers. I decided to duplicate the code (rather than merging the files) since the other functions in this file are not applicable to Realm Objective-C's tests.